.github/workflows/static-checks.yml

```diff
@@ -51,12 +51,33 @@ jobs:
         with:
           distribution: 'zulu'
           java-version: ${{ matrix.java }}
-          cache: 'maven'
+
+      - name: restore maven dependencies
+      # restore maven dependencies only. 'depsonly' keys are safe to restore
+      # since they don't contain any artifacts created by mvn install
+      - uses: actions/cache/restore@v3
+        with:
+          key: maven-${{ runner.os }}-depsonly-${{ hashFiles('**/pom.xml') }}
+          path: ~/.m2/repository
+          restore-keys: |
+            maven-${{ runner.os }}-depsonly-
+
+      - name: compile and cache maven dependencies
+        # run maven compile step to cache all shared maven dependencies
+        # maven install will be run in a subsequent step to avoid polluting the global maven cache
```

Comment:
I wonder why you haven't replied back to my comments;
why can't we just exclude the artifacts built by the project from the cache?

Reply:
excluding artifacts is brittle since it requires knowing every artifact that the project might write to the local maven repo, and knowing what might or might not be safe to cache.

Instead I'd rather we err on the safe side and only add things we know are safe to cache.

Reply:
I believe we are developing the project ...and it should only produce `org.apache.druid` artifacts - if it doesn't we should fix that...
but all of them seem to be under `.m2/repository/org/apache/druid` - do you know about any which is not installed there?

I don't feel that using a crafted maven command to try to warm up the cache would be a better approach

Reply:
if we also want to be able to cache things like npm for the web console, then we will need to rely on a compile/test-compile step to get all the cache entries warmed up. Without that you cannot guarantee that all plugins will be downloaded and invoked to pull their dependencies